quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java

```diff
@@ -47,77 +45,83 @@
-        String[] bearerToken = accessToken.split("Bearer ");
-        UserAuthTokenEntity userAuthTokenEntity = createQuestionBusinessService.verifyAuthToken(bearerToken[1]);
-        QuestionEntity questionEntity = new QuestionEntity();
-        questionEntity.setUuid(UUID.randomUUID().toString());
-        questionEntity.setUser(userAuthTokenEntity.getUser());
+    public ResponseEntity<QuestionResponse> createQuestion(final QuestionRequest questionRequest, @RequestHeader("authorization") final String authorization) throws AuthorizationFailedException {
+
+        // Create question entity
+        final QuestionEntity questionEntity = new QuestionEntity();
         questionEntity.setContent(questionRequest.getContent());
-        final ZonedDateTime now = ZonedDateTime.now();
-        questionEntity.setDate(now);
-        final QuestionEntity createdQuestionEntity = createQuestionBusinessService.createQuestion(questionEntity);
+        questionEntity.setUuid(UUID.randomUUID().toString());
+        questionEntity.setDate(ZonedDateTime.now());
+
+        // Return response with created question entity
+        final QuestionEntity createdQuestionEntity = createQuestionBusinessService.createQuestion(questionEntity, authorization);
         QuestionResponse questionResponse = new QuestionResponse().id(createdQuestionEntity.getUuid()).status("QUESTION CREATED");
         return new ResponseEntity<QuestionResponse>(questionResponse, HttpStatus.CREATED);
     }
 
     @RequestMapping(method = RequestMethod.GET, path = "/question/all", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
-    public ResponseEntity<List<QuestionDetailsResponse>> getAllQuestions(@RequestHeader("accessToken") final String accessToken) throws AuthorizationFailedException {
-        String[] bearerToken = accessToken.split("Bearer ");
-        UserAuthTokenEntity userAuthTokenEntity = authorizationService.verifyAuthToken(bearerToken[1], EndPointIdentifier.QUESTION_ENDPOINT);
-        List<QuestionEntity> questionEntityList = getAllQuestionsBusinessService.getAllQuestions(bearerToken[1]);
-        List<QuestionDetailsResponse> questionDetailsResponseList = new ArrayList<QuestionDetailsResponse>();
-        if (!questionEntityList.isEmpty()) {
-            for (QuestionEntity questionEntity : questionEntityList) {
-                QuestionDetailsResponse questionDetailsResponse = new QuestionDetailsResponse();
-                questionDetailsResponse.setId(questionEntity.getUuid());
-                questionDetailsResponse.setContent(questionEntity.getContent());
-                questionDetailsResponseList.add(questionDetailsResponse);
-            }
+    public ResponseEntity<List<QuestionDetailsResponse>> getAllQuestions(@RequestHeader("authorization") final String authorization) throws AuthorizationFailedException {
+
```

Comment:
As per the instruction we are supposed to do this also.
I mean Authentication verification and proper messages.

Like this
UserAuthTokenEntity userAuthTokenEntity = authorizationService.verifyAuthToken(bearerToken[1], EndPointIdentifier.QUESTION_ENDPOINT);
        List<QuestionEntity> questionEntityList = getAllQuestionsBusinessService.getAllQuestions(bearerToken[1]);

Also we give Access token in the form of   "Bearer <Accesstoke>" , We need to have the logic to handle that here.